rust/otap-dataflow/crates/engine/src/pipeline_ctrl.rs

```diff
@@ -103,9 +103,6 @@ impl PipelineCtrlMsgManager {
                         if when > now {
                             tokio::time::sleep_until(when).await;
                         }
-                    } else {
```

Comment:
This `else` block is unnecessary since we already have a guard for this select branch:
```rust
if next_expiry.is_some()
```